parts/k8s/windowsantreacnifunc.ps1

```diff
@@ -0,0 +1,42 @@
+function Install-OpenvSwitch
+{
+    Param(
+        [string]
+        $OVSUrl = "https://raw.githubusercontent.com/vmware-tanzu/antrea/master/hack/windows/Install-OVS.ps1",
+        [Parameter(Mandatory = $true)][string]
+        $KubeDir
+    )
+
+    # Enable TESTSIGNING so that OVS datapath can be enabled. This settings is not recommended for production
+    & Bcdedit.exe -set TESTSIGNING ON
+
+    # TODO: Discuss with AKS Folks, how to install Upstream OVS on Windows Node
+    # and sign it.
```

Comment:
Can you explain more on the requirement for signing the ovs? For aks-engine users will they be able use this as is or is there another process that is required for them to get signed binaries?

Reply:
Antrea is using the Open Source OVS as the forwarding component, nd the Open Source OVS has no certificate from Microsoft. Hence, we have to enable the "testsigning" mode on the Windows host to ensure OVS datapath could forward packets correctly. 

But there might be some security risks to enable the "testsigning" mode on a Windows host, so we want to get more thoughts from the AKS folks to see if we can we get signed for the Open Source OVS (then we don't need to enable the testsigning mode), or the risk for the "testsigning" mode is acceptable (then we could continue using the unsigned Open Source OVS).

Reply:
I don't have any experience with OVS so have a few more questions: 

- For folks who don't use AKS and only use aks-engine but would like to use OVS how will they get the signed cert? 
- Do they need to rely on Microsoft to provide a certificate?  Will they want to provide a cert that they sign?  
- In the case of AKS, will customers also want to provide there own certs?

Want to make sure there is a path for teams and users that don't use AKS as well.

@AbelHu might have some additional questions or insight as well.

Reply:
Without a Microsoft official certificate, "testsigning" mode must be enabled even if the user generates a self-signed cert.

To acquire an official certificate, we need an organization. However, either the open source OVS community or the Linux Foundation doesn't provide the signed cert currently. So the OVS community recommends users to enable "testsigning" mode when using OVS on Windows host.

An official certificate requires all the binaries in OVS should be signed that also includes the installers, catalog files and driver files. The action owner should be an orgnization first, and has an account in Microsoft to trigger the signing process. A series of tests playlist that will run against some nodes which already have OVS deployed. Once all the tests have been passed, a signed package can be created and uploaded to Microsoft for verification and further AV scanning.

Reply:
so the binaries need to be signed and there isn't a cert that needs to be provided? This means Microsoft could sign the binaries out of band and distribute them many customers?  We also require our binaries and powershell scripts to be signed (#3441 for example)

I had initially thought there was a requirement for a private cert that needed to be installed on the node per customer.

Reply:
Yes @jsturtevant. If Microsoft can sign the binary and put in under Azure storage, we can update our install script to fetch the binary from that location. 
Also, it is possible to consider a new Windows VHD from Microsoft with OVS already installed on it?

Reply:
Thanks this makes sense. We do something similar for other packages.  Given this isn't GA yet we can leave the self signing for folks to try it out. As it matures and demand grows we can address signing.  We will want to make sure that the components are not installed unless the cni is installed this way it doesn't get flagged for signing by any processes that might be checking.

Reply:
Hi @jsturtevant , the OVS is signed by Microsoft means WHQL certification for OVS binaries.

Following arr the steps to get the WHQL certification:

1. You need to pass the tests for HLK on Windows 10 family and above
(https://docs.microsoft.com/en-us/windows-hardware/test/hlk/) and
HCK on Windows 8 family
(https://docs.microsoft.com/en-us/windows-hardware/drivers/develop/run-the-hck-test-suites-in-the-wdk).
The above is a series of tests playlist that will be ran against some nodes which already have OVS deployed.
The tests are a bit finicky (to say at best) and vary between versions and types (Server / Client – Home, Professional
and Enterprise) of Windows OS.

2. Once you pass all the tests you create a signed package which you upload to Microsoft for verification and
further AV scanning.

To note: The driver certificate that you use on the deployed nodes prior to HCK/HLK testing, has to be signed
using a different, more expensive certificate: EV certificate
(https://docs.microsoft.com/en-us/windows-hardware/drivers/dashboard/get-a-code-signing-certificate)
I’m not sure if they dropped this requirement but it was a huge issue when people wanted to sign drivers
for 8.1/2012 R2 and 10-RS4/ Server 2016.

Again an organization has to be behind that certificate.

WHQL signing means that Microsoft will cross sign the binaries that we pushed in our publisher package.
This will also allow the use of Secure Boot.

Reply:
When we address the signing aspect as discussed above we will also need include signing the `AntreaStartWrapper` and `https://github.com/vmware-tanzu/antrea/releases/download/v0.9.0/Start.ps1`.

Reply:
I'm not sure if the aks-engine team can perform the type of signing needed here.
We currently perform authenicode signing for many open source components (mainly kubernetes binaries) but that is intended to provide assurance that these binaries were built by Microsoft and not modified.
We don't currently have the capabilities to sign with certificates that allow code / driver execution.

Reply:
Also, enabling test signing on AKS will not be acceptable.
We'll need to figure out how to get OVS to be signed properly.